backend/auto_deploy_migration.py

```
@@ -294,6 +297,64 @@ async def run_auto_deployment_migrations():
+        
+        if source_url_exists:
+            # Perform the backfill with truncation
+            rows_updated = await conn.fetchval("""
+                UPDATE public.rag_knowledge_base 
+                SET source_reference = LEFT(source_url, 255) 
+                WHERE source_reference IS NULL AND source_url IS NOT NULL
+                RETURNING (SELECT COUNT(*) FROM public.rag_knowledge_base 
+                           WHERE source_reference IS NOT NULL AND source_url IS NOT NULL)
+            """)
+            
+            logger.info(f"✅ Source reference backfill completed. Updated rows from source_url.")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)